src/tuesday/core/plotting/powerspectra.py

```diff
@@ -0,0 +1,389 @@
+        cmap=cmap,
+        **cmap_kwargs,
+    )
+
+    ax.set_xlabel(xlabel, fontsize=fontsize)
+    if cbar:
+        plt.colorbar(im, label=clabel)
+    if log[0]:
+        ax.set_xscale("log")
+    if log[1]:
+        ax.set_yscale("log")
+
+    return fig, ax
+
+
+def plot_power_spectrum(
+    wavemodes: un.Quantity,
+    power_spectrum: un.Quantity,
+    fig: plt.Figure | None = None,
+    ax: plt.Axes | list[plt.Axes] | None = None,
+    title: str | None = None,
+    xlabel: str | None = None,
+    ylabel: str | None = None,
+    clabel: str | None = None,
+    cmap: str | None = "viridis",
+    color: list | None = None,
+    fontsize: float | None = 16,
+    vmin: float | None = None,
+    vmax: float | None = None,
+    log: list[bool] | None = False,
+    cbar: bool | None = True,
+    label: list | None = None,
+    smooth: float | bool = False,
+    legend_kwargs: dict | None = None,
+) -> tuple[plt.Figure, plt.Axes]:
+    """
+    Plot the power spectrum.
+
+    Parameters
+    ----------
+    wavemodes : np.ndarray
+        Wavemodes corresponding to the power spectrum.
+    power_spectrum : np.ndarray
+        Power spectrum values.
+    fig : plt.Figure, optional
+        Figure object to plot on. If None, a new figure is created.
+    ax : plt.Axes | list[plt.Axes], optional
+        Axes object(s) to plot on. If None, new axes are created.
+    title : str, optional
+        Title of the plot.
+    xlabel : str, optional
+        Label for the x-axis.
+    ylabel : str, optional
+        Label for the y-axis.
+    clabel : str, optional
+        Label for the colorbar.
+    cmap : str, optional
+        Colormap for the plot.
+    colors : list, optional
+        List of colors for each line in the plot.
+    fontsize : float, optional
+        Font size for the plot labels.
+    vmin : float, optional
+        Minimum value for the color scale.
+    vmax : float, optional
+        Maximum value for the color scale.
+    log : list[bool], optional
+        List of booleans to set the axes to log scale.
+    labels : list, optional
+        List of labels for each line in the plot.
+    smooth : bool or float, optional
+        Standard deviation for Gaussian smoothing.
+        If True, uses a standard deviation of 1.
+    legend_kwargs : dict, optional
+        Keyword arguments for the legend.
+    """
+    power_spectrum = power_spectrum.squeeze()
+    if isinstance(label, str):
+        label = [label]
+    if not np.iterable(log):
+        log = [log]
+    if (hasattr(wavemodes, "ndim") and wavemodes.ndim == 1) or len(wavemodes) == 1:
+        if legend_kwargs is None:
+            legend_kwargs = {}
+        if len(log) == 1:
+            log = [log[0], log[0]]
+        if power_spectrum.ndim > 1:
+            raise ValueError("Plot one 1D PS at a time.")
+        if fig is None:
+            fig, ax = plt.subplots(1, 1, figsize=(8, 6))
+        if ax is None:
+            ax = fig.get_axes()[0]
```

Comment:
I think actually its cleaner for the user just to be able to pass `ax`, and then you can call `ax.get_figure()` to get the figure its on. So no `fig` argument.